File: table-order/services/order-service/app/models/order.py
from dataclasses import dataclass
from datetime import datetime


@dataclass
class Order:
    id: int
    store_id: int
    table_id: int
    session_id: int
    order_number: str
    status: str  # "pending" | "preparing" | "completed"
    total_amount: int
    created_at: datetime
    updated_at: datetime

File: table-order/frontend/packages/shared/src/types/order.ts
export type OrderStatus = 'pending' | 'preparing' | 'completed';

export interface Order {
  id: string;
  storeId: string;
  tableId: string;
  sessionId: string;
  orderNumber: string;
  status: OrderStatus;
  totalAmount: number;
  items: OrderItem[];
  createdAt: string;
  updatedAt: string;
}

export interface OrderItem {
  id: string;
  orderId: string;
  menuName: string;
  quantity: number;
  unitPrice: number;
  options: OrderItemOption[];
  subtotal: number;
}

export interface OrderItemOption {
  id: string;
  orderItemId: string;
  optionName: string;
  additionalPrice: number;
}

export type OrderEvent =
  | { type: 'new_order'; data: Order }
  | { type: 'status_changed'; data: { orderId: string; status: OrderStatus } }
  | { type: 'order_deleted'; data: { orderId: string; tableId: string } };
